desktop/apps/consignments2/client/actions.js

```diff
@@ -328,3 +404,72 @@ export function updateSubmission (submission) {
     }
   }
 }
+
+export function uploadImageToConvection (filePath, fileName) {
+  return async (dispatch, getState) => {
+    try {
+      const token = await fetchToken()
+      const { submissionFlow: { submission } } = getState()
+      const geminiToken = last(filePath.split('/')).split('%2F')[0] // FIXME get the gemini token another way
```

Comment:
I want to look into this further, too. Gemini returns a `token` when you initially request credentials, it just doesn't appear to be passed through `gemup`. Hoping I can update `gemup` to pass it through and not have to do this url-splitting.